[docs]: Clarify vendored sources as read-only and way to modify (#13512)
rust-lang/cargo: src/doc/man/cargo-vendor.md

You will need to add or redirect it to your Cargo configuration file,
which is usually `.cargo/config.toml` locally for the current package.

> By default, Cargo treats vendored crates as read-only as it does crates.io.

Reviewer: “By default” sounds like there is a way to tweak it, though there is no way to change the read-only property.

```suggestion
Cargo treats vendored sources as read-only as it does to registry and git sources.
```

> To modify a vendored crate the correct way is to use `[patch]` or a `path` dependency,
> and Cargo will then correctly handle the crate on incremental rebuilds as it knowns
> that it is not a read-only dependency.

Reviewer: A few tweaks. What do you think?

```suggestion
If you intend to modify a crate from a remote source,
use `[patch]` or a `path` dependency pointing to a local copy of that crate.
Cargo will then correctly handle the crate on incremental rebuilds,
as it knowns that it is no longer a read-only dependency.
```

Author: Thanks, this looks more accurate.

You will need to add or redirect it to your Cargo configuration file,
which is usually `.cargo/config.toml` locally for the current package.

> Cargo treats vendored sources as read-only as it does to registry and git sources.  

Reviewer: Just notice there are some trailing spaces. Not sure if it is intentionally done but they do render to a linebreak here. Otherwise looks good to me.

```suggestion
Cargo treats vendored sources as read-only as it does to registry and git sources.
```

Author: More thanks, modified
